content/docs/glossary/claimable-balance.mdx

```diff
@@ -49,12 +49,57 @@ This operation will load the ClaimableBalanceEntry that corresponds to the Balan
 Once a ClaimableBalanceEntry has been claimed, it will be deleted.
 
 ## Example
-The below code uses the [Go SDK](../software-and-sdks/index.mdx) to demonstrate how an account ("Account A") can create a `ClaimableBalanceEntry` with two Claimants, Account A (itself) and "Account B", another recipient.
+The below code demonstrates via both the JavaScript and Go [SDKs](../software-and-sdks/index.mdx) how an account ("Account A") can create a `ClaimableBalanceEntry` with two Claimants: Account A (itself) and "Account B" (another recipient).
 
 Each of these accounts can only claim the balance under certain, individual conditions. Namely, Account B has a full minute to claim the balance, after which Account A can "reclaim" the balance back for itself.
 
+It's worth emphasizing that there is no "recovery" mechanism for a claimable balance in general: if none of the predicates can be fulfilled, the balance **cannot be recovered**. The "reclaim" paradigm below acts as a safety net for this situation.
+
 <CodeExample>
 
+```js
+const sdk = require('stellar-sdk');
+
+var server = new sdk.Server("https://horizon-testnet.stellar.org")
+
+var A = sdk.Keypair.fromSecret("SAQLZCQA6AYUXK6JSKVPJ2MZ5K5IIABJOEQIG4RVBHX4PG2KMRKWXCHJ")
+var B = sdk.Keypair.fromPublicKey("GAS4V4O2B7DW5T7IQRPEEVCRXMDZESKISR7DVIGKZQYYV3OSQ5SH5LVP")
+
+// NOTE: Error checks are omitted for brevity; always validate things!
+
+server
```

Comment:
I think we prefer to use `await` instead of chaining promises

Reply:
Personally I prefer Promises over Async/Await but we should stick to whatever is the most commonly used pattern in our official documentations.

Reply:
Tangentially-related to this: what's the best way to pass along multiple values from Promise to Promise? I was doing something like:
```js
  return [a, b]
}).then(function(pair) {
  let a = pair[0], b = pair[1]
  // the rest...
})
```
but that felt *really* odd to me. Are there better ways?

The reason I ask, for additional context, is because sometimes you want to use long-term values (like accounts) in addition to shorter-term values (like a transaction) in a single Promise body. Or is this already a flawed way of thinking?

Reply:
In typescript you can return a typed object, but not so sure for javascript, it's been a few years since I did some heavy JS work.